multichain-testing/tools/query.ts

```diff
@@ -21,14 +21,25 @@ type BlockHeaderPartial = {
 // TODO: inject fetch
 export function makeQueryClient(apiUrl: string) {
   const query = async <T>(path: string): Promise<T> => {
-    try {
-      const res = await fetch(`${apiUrl}${path}`);
-      const json = await res.json();
-      return json as T;
-    } catch (err) {
-      console.error(err);
-      throw err;
-    }
+    const maxRetries = 5;
+    const retryDelayMS = 500;
```

Comment:
don't we have a general "retry promise up to N times with M delay" utility? cc @dckc

Reply:
We have `retryUntilCondition` but there's no condition to check here, just that the request worked, so it felt strange to use here. This is just to stop the test from flaking when the RPC connection fails which I observed a few times.

Reply:
That the request worked sounds like a condition. If it's hard to use that function for this case, maybe we could use another. Food for thought.